### ISSUE TITLE
askama workspace build problem

### ISSUE THREAD

**Issue by johanot** [cargo2nix/cargo2nix#291]
When trying to build e.g. askama_escape (https://github.com/djc/askama/blob/main/askama_escape/Cargo.toml) with cargo2nix, I get `Cargo.toml: No such file or directory`. I believe it happens because Cargo looks for the workspace root Cargo.toml which has been moved/renamed during the buildPhase.

Only when patching out https://github.com/djc/askama/blob/main/askama_escape/Cargo.toml#L10 (`workspace = ".."`) things build correctly.

Forgive my lack of knowledge on workspace builds. Am I doing something wrong or are the askama authors using the workspace directive wrong? Is cargo2nix lacking something here?

**Comment by psionic-k** [comment]
Could just be the first time that an explicit workspace came up.  I'll make sure that the TOML used in the build phase doesn't try to look up to a workspace that doesn't exist.

**Comment by psionic-k** [comment]
Documenting how to adjust the retained cargo manifest tables / values for future readers:

https://github.com/cargo2nix/cargo2nix/blob/release-0.11.0/overlay/mkcrate.nix#L240-L250

This jq query needs to filter out attributes that interfere with the builder's behavioral intent to build each crate isolation.

The workspace attribute is one such [cargo manifest](https://doc.rust-lang.org/cargo/reference/manifest.html) value.

To fix issues like this, add attributes that break the crate, and then configure a query that will successfully remove the offending attribute.   In this case, I pointed the cargo workspace to nowhere and then verified that nix build succeeds by removing the erroneous manifest value.

https://github.com/cargo2nix/cargo2nix/pull/292/commits/e95e9601257602bbec8c05a18edf114855f0efc5

Fix will be pushed to unstable after CI passes.  Closing.
